torch/csrc/jit/tensorexpr/tensorexpr_init.cpp

```diff
@@ -430,6 +431,54 @@ void initTensorExprBindings(PyObject* module) {
       [](Stmt* stmt) { return IRSimplifier::simplify(stmt); },
       py::return_value_policy::reference);
 
+  using TSGraph = std::shared_ptr<Graph>;
+  py::class_<TensorExprKernel>(te, "TensorExprKernel")
+      .def(py::init<const TSGraph>())
+      .def(
+          "run",
+          [](TensorExprKernel& self, const py::tuple& inputs) {
+            Stack stack;
+            stack.reserve(inputs.size()); // captures?
+            for (auto& obj : inputs) {
+              stack.push_back(toTypeInferredIValue(obj));
+            }
+            auto g_inputs = self.graph()->inputs();
+            for (size_t i = 0; i < inputs.size(); ++i) {
+              if (stack[i].isTensor()) {
+                g_inputs[i]->setType(stack[i].type());
+              }
+            }
+            self.run(stack);
+            return createPyObjectForStack(std::move(stack));
+          })
+      .def(
+          "fallback",
+          [](TensorExprKernel& self, const py::tuple& inputs) {
+            Stack stack;
+            stack.reserve(inputs.size()); // captures?
+            for (auto& obj : inputs) {
+              stack.push_back(toTypeInferredIValue(obj));
+            }
+            auto g_inputs = self.graph()->inputs();
+            for (size_t i = 0; i < inputs.size(); ++i) {
+              if (stack[i].isTensor()) {
+                g_inputs[i]->setType(stack[i].type());
+              }
+            }
+            self.fallback(stack);
+            return createPyObjectForStack(std::move(stack));
+          })
+      .def(
+          "get_code_gen_stmt",
```

Comment:
Nit: I think we usually use `codegen` without underscore in the middle, let's stay consistent with it.